src/main/kotlin/com/sourcegraph/cody/agent/CodyAgentClient.kt

```diff
@@ -253,4 +260,21 @@ class CodyAgentClient(private val webview: NativeWebviewProvider) {
 
     logger.debug("webview/postMessage ${params.id}: ${params.message}")
   }
+
+  @JsonRequest("window/showSaveDialog")
+  fun window_showSaveDialog(): CompletableFuture<String> {
```

Comment:
`showSaveDialog` in VSC have quite a few optional parameters which we should respect:

```
	export interface SaveDialogOptions {
		/**
		 * The resource the dialog shows when opened.
		 */
		defaultUri?: Uri;

		/**
		 * A human-readable string for the save button.
		 */
		saveLabel?: string;

		/**
		 * A set of file filters that are used by the dialog. Each entry is a human-readable label,
		 * like "TypeScript", and an array of extensions, e.g.
		 * ```ts
		 * {
		 * 	'Images': ['png', 'jpg']
		 * 	'TypeScript': ['ts', 'tsx']
		 * }
		 * ```
		 */
		filters?: { [name: string]: string[] };

		/**
		 * Dialog title.
		 *
		 * This parameter might be ignored, as not all operating systems display a title on save dialogs
		 * (for example, macOS).
		 */
		title?: string;
	}
```

Reply:
These options are always `undefined`
![image](https://github.com/user-attachments/assets/2cb266c2-96fe-4d14-8ff7-c304da3e5465)